hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/metrics/TestSystemMetricsPublisherInBatch.java

```diff
@@ -0,0 +1,62 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.yarn.server.resourcemanager.metrics;
+
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+import org.apache.hadoop.yarn.server.applicationhistoryservice.ApplicationHistoryServer;
+import org.apache.hadoop.yarn.server.timeline.MemoryTimelineStore;
+import org.apache.hadoop.yarn.server.timeline.TimelineStore;
+import org.apache.hadoop.yarn.server.timeline.recovery.MemoryTimelineStateStore;
+import org.apache.hadoop.yarn.server.timeline.recovery.TimelineStateStore;
+import org.junit.BeforeClass;
+
+public class TestSystemMetricsPublisherInBatch extends TestSystemMetricsPublisher {
+
+  private static ApplicationHistoryServer timelineServer;
+  private static TimelineServiceV1Publisher metricsPublisher;
+  private static TimelineStore store;
+
+  @BeforeClass
```

Comment:
I think `@Override` can be added for better readability.

Reply:
It's a static method, so I think we cant override it.

Reply:
Oh, sorry for that. Please ignore.

Reply:
Rethinking this, the TestSystemMetricsPublisherInBatch runs TestSystemMetricsPublisher#setup() method and then runs TestSystemMetricsPublisherInBatch#setup(), so I suppose there are two TimelineServers running in the test.
Probably we can use JUnit parameterized test instead of extending the test class. https://github.com/junit-team/junit4/wiki/parameterized-tests

@ashutoshcipher what do you think?

Reply:
Yes, we can do that. I have updated the current test. You can have a look.